hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -871,38 +869,11 @@ certificateStore, null, new DefaultCAProfile(),
           CAType.SUBORDINATE);
     }
 
-    // If primary SCM node Id is set it means this is a cluster which has
-    // performed init with SCM HA version code.
-    if (scmStorageConfig.checkPrimarySCMIdInitialized()) {
```

Comment:
I believe we should not remove this code piece.

Ozone 1.2 introduced SCM HA, but we should enable the upgrade from Ozone 1.1 with a single SCM node to Ozone 1.4 probably, and this code part deals with the things required to recoginize that we are after such an upgrade, set the cluster up that way, and make it possible to switch to SCM HA after the upgrade.

I believe this is something which we should keep at least until we do a major version change, as this is an incompatible change we should not do it in a minor release.

Also in such an upgrade scenario, we already have the keys on disk, and we need the rootCAServer as we create the sub-CA certificates during the first steps after the upgrade happened, and for that we need the rootCA server.